src/mechanics/ingame/demonslayer/giyu/TomiokaGiyu.java

```diff
@@ -1,37 +1,82 @@
 package mechanics.ingame.demonslayer.giyu;
 
+import mechanics.Texts;
 import mechanics.abilities.Ability;
 import mechanics.characters.Character;
 
 public class TomiokaGiyu extends Character {
+	
+	private int instantKillBoostFromAbility1;
+	private int instantKillBoostFromAbility2;
 
 	public TomiokaGiyu(int id) {
 		super(id);
-		// TODO Auto-generated constructor stub
+		
+		this.instantKillBoostFromAbility1 = 0;
+		this.instantKillBoostFromAbility2 = 0;
+	}
+	
+	private void clearInstantKillBuffs() {
+		this.removeActive(this.getA1());
+		this.removeActive(this.getA2());
```

Comment:
This one is my fault, for reviewing it wrongly. 
When you apply the 3rd Ability, you remove the active of the 1st and 2nd. That is correct and should happen, but the way I suggested doesnt work for this situation.

After you apply an Ability, a new object is created. So imagine for this Character, you apply the 1st Ability. Then use the 3rd Ability. Because a new 1st Ability was created, you are not removing the 1st Ability that was used, but the new one instead (and the new one has nothing to be removed). 

So basically, its necessary to store the 1st & 2nd Ability that boosted the 3rd ability effect, and remove those abilities.

A suggestion of mine is to store it on a list, as an attribute of the class:
`List<Ability> instantKillBoostAbilities = new ArrayList<>();`

Then here, we loop the list and reset it:
```suggestion
		for (Ability ability : instantKillBoostAbilities) {
			this.removeActive(ability);
		}
		this.instantKillBoostAbilities.clear();
```